src/engraving/rendering/dev/beamlayout.cpp

```diff
@@ -96,16 +96,34 @@ void BeamLayout::layout(Beam* item, LayoutContext& ctx)
         }
         layout2(item, ctx, crl, st, static_cast<int>(n));
 
-        double lw2 = item->beamWidth() / 2.0;
-
-        for (const BeamSegment* bs : item->beamSegments()) {
-            PolygonF a(4);
-            a[0] = PointF(bs->line.x1(), bs->line.y1());
-            a[1] = PointF(bs->line.x2(), bs->line.y2());
-            a[2] = PointF(bs->line.x2(), bs->line.y2());
-            a[3] = PointF(bs->line.x1(), bs->line.y1());
-            RectF r(a.boundingRect().adjusted(0.0, -lw2, 0.0, lw2));
-            item->addbbox(r);
+        if (item->staffType() && item->staffType()->isJianpu()) {
+            double symWidth = item->symWidth(SymId::keysigJianpuDo);
+            double symHeight = item->symHeight(SymId::keysigJianpuDo);
+            double lw2 = item->beamWidth() / 6.0;
+            for (const BeamSegment* bs : item->beamSegments()) {
+                double y1 = bs->line.y1() / 1.65 + symHeight / 2.55;
```

Comment:
Another one... if you have to divide by `2.55`, that looks to me like there should actually not be a direct relation between the height of that symbol and the value you want to calculate here. Instead, you could base the value on the overall size of items in the score (i.e. `spatium`), or create a style setting for it.

(regarding style settings: I mean those parameters that can be accessed using keys from the  `Sid` enum. Not all of them are (currently) exposed as an actual style setting in the UI, but the `Sid` system is useful for forcing yourself to use named constants, and makes it easy to make certain values still customisable in the future.)